src/graph/graph.c

```diff
@@ -867,17 +843,50 @@ void Graph_GetNodeEdges
 	ASSERT(g);
 	ASSERT(n);
 	ASSERT(edges);
+	ASSERT(edgeType != GRAPH_NO_RELATION && edgeType != GRAPH_UNKNOWN_RELATION);
 
-	GrB_Type t;
 	GrB_Info info;
-	RG_MatrixTupleIter   it       =  {0};
-	RG_Matrix            M        =  NULL;
-	RG_Matrix            TM       =  NULL;
-	NodeID               srcID    =  ENTITY_GET_ID(n);
-	NodeID               destID   =  INVALID_ENTITY_ID;
-	EdgeID               edgeID   =  INVALID_ENTITY_ID;
+	Delta_MatrixTupleIter   it       =  {0};
+	Delta_Matrix            S        =  NULL;
+	Delta_Matrix            T        =  NULL;
+	NodeID                  src_id   =  ENTITY_GET_ID(n);
```

Comment:
wouldn't you consider `n` as the destination node? we're searching for incoming edges
`(n)<-(s)`